hardware_interface/include/hardware_interface/resource_manager.hpp

```diff
@@ -0,0 +1,115 @@
+// Copyright 2020 ROS2-Control Development Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef HARDWARE_INTERFACE__RESOURCE_MANAGER_HPP_
+#define HARDWARE_INTERFACE__RESOURCE_MANAGER_HPP_
+
+#include <map>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "hardware_interface/actuator_hardware_interface.hpp"
+#include "hardware_interface/actuator_hardware.hpp"
+#include "hardware_interface/robot_hardware_interface.hpp"
+#include "hardware_interface/sensor_hardware_interface.hpp"
+#include "hardware_interface/sensor_hardware.hpp"
+#include "hardware_interface/system_hardware_interface.hpp"
+#include "hardware_interface/system_hardware.hpp"
+#include "hardware_interface/types/hardware_interface_return_values.hpp"
+#include "hardware_interface/visibility_control.h"
+#include "pluginlib/class_loader.hpp"
+
+using hardware_interface::return_type;
+
+namespace resource_manager
+{
+
+class ResourceManager
+{
+public:
+  HARDWARE_INTERFACE_PUBLIC
+  ResourceManager();
+
+  HARDWARE_INTERFACE_PUBLIC
+  virtual
+  ~ResourceManager() = default;
+
+  //  Non real-time safe functions
+  HARDWARE_INTERFACE_PUBLIC
+  return_type load_and_configure_resources_from_urdf(std::string urdf_string);
+
+  HARDWARE_INTERFACE_PUBLIC
+  return_type check_command_interfaces(
+    const std::string & joint_name, const std::vector<std::string> & interfaces) const;
```

Comment:
why are these public functions?

Reply:
They are currently used to check if the command interfaces exist and there is no conflict. They're called from the controller. There was similar implementation in robot hardware.

This can be removed if we decide to go for #183